AWSPinpoint/AWSPinpointNotificationManager.m

```diff
@@ -257,4 +260,73 @@ - (void)addApplicationStateAttributeToEvent:(AWSPinpointEvent *) event
     }
 }
 
+- (AWSPinpointEvent*)buildEventFromUserInfo:(NSDictionary *) userInfo
+                         withPushActionType:(AWSPinpointPushActionType) pushActionType {
+    NSString *eventTypePrefix = [self getEventTypePrefixFromUserInfo:userInfo];
+    if (!eventTypePrefix) {
+        return nil;
+    }
+    NSString *eventTypeSuffix = [self getEventTypeSuffixFromPushActionType:pushActionType];
+    NSString *eventType = [NSString stringWithFormat: @"_%@.%@", eventTypePrefix, eventTypeSuffix];
+    AWSPinpointEvent *pushNotificationEvent  = [self.context.analyticsClient createEventWithEventType:eventType];
+    [self addApplicationStateAttributeToEvent:pushNotificationEvent
+                         withApplicationState:[[UIApplication sharedApplication] applicationState]];
+    return pushNotificationEvent;
+}
+
+- (NSString*)getEventTypePrefixFromUserInfo:(NSDictionary *) userInfo {
+    if ([AWSPinpointNotificationManager validPinpointPushForNotification:userInfo]) {
+        NSDictionary *pinpointData = userInfo[AWSDataKey][AWSPinpointKey];
+        if (pinpointData[AWSPinpointCampaignKey]) {
+            return AWSPinpointCampaignKey;
+        } else if (pinpointData[AWSPinpointJourneyKey]) {
+            return AWSPinpointJourneyKey;
+        }
+    }
+    AWSDDLogError(@"Cannot determine event type from Push payload");
+    return nil;
+}
+
+- (NSString*)getEventTypeSuffixFromPushActionType:(AWSPinpointPushActionType) pushActionType {
+    switch (pushActionType) {
+        case AWSPinpointPushActionTypeOpened:
+            return AWSEventTypeOpened;
+        case AWSPinpointPushActionTypeReceivedForeground:
+            return AWSEventTypeReceivedForeground;
+        case AWSPinpointPushActionTypeReceivedBackground:
+            return AWSEventTypeReceivedBackground;
+    }
+}
```

Comment:
Generally i favor favor reducing the number of exit points in a function, so i prefer something liek:
```
    NSString * eventType;
    switch (pushActionType) {
        case AWSPinpointPushActionTypeOpened:
            eventType = AWSEventTypeOpened;
            break;
        case AWSPinpointPushActionTypeReceivedForeground:
            eventType = AWSEventTypeReceivedForeground;
            break;
        case AWSPinpointPushActionTypeReceivedBackground:
            eventType = AWSEventTypeReceivedBackground;
            break;
    }
    return eventType;
```

Reply:
Will update this and `pushActionTypeOfApplicationState`